### ISSUE TITLE
Autocomplete for PM should work even when someone isn't in the same room

### ISSUE THREAD

**Issue by sethwebster** [JabbR/JabbR#229]
As long as they are signed in.

**Comment by kiliman** [comment]
This is a little tricky to implement. The current implementation gets the user list from the current room. This list is updated when users join, leave, or change nicks. Since the list is local, it is very fast.

However, to get the list of online users would require a call to the server. Although we could make the call at startup, we would then have to maintain this list. Right now, the server only calls the client about changes to users in rooms they are joined in. To make this list accurate, we would need to update ALL clients anytime a new user came online. I'm concerned about the performance impact on something that will be used infrequently.

Also I'd have to figure out a way to return different different user lists if user enters /msg vs normal message.

Any thoughts?

**Comment by sethwebster** [comment]
Couldn't we trap a tab on /msg set<tab> and query the server for the info?

**Comment by davidfowl** [comment]
No, we can keep a global list of users updated on the client or we update it in the background.

**Comment by dbertram** [comment]
As an alternative to pinging the server/keeping a client-side cache of all online users, could we provide autocomplete for the set of users across _all_ the rooms you're currently in?

This keeps things client-side (i.e., fast), but is more powerful than the current implementation.

Here's a scenario (from the company I work for) where this would really help:
- we have a general-purpose chat room that everyone idles in
- I'm in a team-specific room, but want to PM a user that's not in that room (but _is_ in the "lobby" room)
